osquery/config/config.cpp

```diff
@@ -270,53 +281,31 @@ void Config::getPerformanceStats(
   }
 }
 
+void Config::hashSource(const std::string& source, const std::string& content) {
+  std::vector<char> buffer;
+  for (const auto& c : content) {
+    buffer.push_back(c);
+  }
+  hash_[source] = hashFromBuffer(HASH_TYPE_MD5, &buffer[0], buffer.size());
+}
+
 Status Config::getMD5(std::string& hash) {
-  if (hash_.empty()) {
-    std::vector<char> buffer;
-    auto add = [&buffer](const std::string& text) {
-      for (const auto& c : text) {
-        buffer.push_back(c);
-      }
-    };
-    scheduledQueries(
-        [&add, &buffer](const std::string& name, const ScheduledQuery& query) {
-          add(name);
-          add(query.query);
-          add(std::to_string(query.interval));
-          for (const auto& it : query.options) {
-            add(it.first);
-            add(it.second ? "true" : "false");
-          }
-        });
+  if (!isValid()) {
+    return Status(1, "Current config is not valid");
+  }
 
-    auto parsers = Registry::all("config_parser");
-    for (const auto& parser : parsers) {
-      add(parser.first);
-      try {
-        if (parser.second == nullptr || parser.second.get() == nullptr) {
-          continue;
-        }
-        auto plugin =
-            std::static_pointer_cast<ConfigParserPlugin>(parser.second);
-        if (plugin == nullptr || plugin.get() == nullptr) {
-          continue;
-        }
-        std::stringstream ss;
-        pt::write_json(ss, plugin->getData());
-        add(ss.str());
-      } catch (const std::bad_cast& e) {
-        LOG(ERROR) << "Error casting config parser plugin: " << e.what();
-      } catch (const pt::ptree_error& e) {
-        LOG(ERROR)
-            << "Error writing config parser content to JSON: " << e.what();
-      }
+  ReadLock rlock(config_hash_mutex_);
+  std::vector<char> buffer;
+  auto add = [&buffer](const std::string& text) {
```

Comment:
For my clarity, why write `&s.c_str()[0]`? It's the same pointer/type as `::data()` or `c_str()`'s return type: a `const char*`. Taking the reference of the dereferenced first value is a bit overkill and looks complicated when it's not needed as a dangling struct pointer.

Or perhaps I'm missing something here?